packages/website/blog/2024-09-30-typed-linting.md

```diff
@@ -0,0 +1,258 @@
+Knowing the types of pieces of your code allows lint rules to flag for risky behavior specific to certain types.
+The following sections show several of the most common uses for lint rules that rely on type information.
+
+### Unsafe `any`s
+
+The `@typescript-eslint/no-unsafe-*` family of rules checks for risky uses of `any` typed values.
+This is useful because the `any` type can easily slip into code and reduce type safety, despite being allowed by the TypeScript type checker.
+
+For example, the following code that logs a member of an object parsed from a string produces no type errors in type checking.
+`JSON.parse()` returns `any`, and arbitrary property accesses are allowed on values of type `any`.
+
+However, [`@typescript-eslint/no-unsafe-member-access`](/rules/no-unsafe-member-access) would report `[key]` might not be a property on the object:
+
+```ts
+function getDataKey(rawData: string, key: string): string {
+  return JSON.parse(rawData)[key];
+  //                        ~~~~~
+  // Unsafe member access [key] on an `any` value.
+  // eslint(@typescript-eslint/no-unsafe-member-access)
+}
+```
+
+The lint rule is right to report.
+Calls to the `getDataKey` function can return a value that's not a `string`, despite the function's explicit return type annotation.
+That can lead to unexpected behavior at runtime:
+
+```ts
+console.log(getDataKey(`{ "blue": "cheese" }`, 'bleu').toUpperCase());
+// Uncaught TypeError: Cannot read properties of undefined (reading 'toUpperCase')
+```
+
+Without type information to indicate the types of `JSON.parse` and `key`, there would have been no way to determine that the `[key]` member access was unsafe.
+
+### Method Call Scoping
+
+<!-- - Catching [dangerous calls to unbound methods that fail to provide a `this`](/rules/no-unbound-method) -->
+
+Runtime crashes caused by misuses of typed code are possible even with no `any`s.
+
+For example, class method functions don't preserve their class scope when passed as standalone variables ("unbound").
+TypeScript still allows them to be called without the proper `this` scope.
+
+The global `localStorage` object in browsers has several properties that must be called with a `this` bound to `localStorage`.
+The [`@typescript-eslint/unbound-method`](/rules/unbound-method) lint rule can report on unsafe references to those properties, such as accessing `getItem`:
+
+```ts
+const { getItem } = localStorage;
+//      ~~~~~~~
+// Avoid referencing unbound methods which may cause unintentional scoping of `this`.
+// eslint(@typescript-eslint/unbound-method)
+```
+
+That's useful because calls to `getItem` that aren't bound to `localStorage` cause an exception at runtime:
+
+```ts
+getItem('...');
+// Uncaught TypeError: Illegal invocation
+```
+
+Without type information to indicate the types of `localStorage` and its `getItem` property, there would have been no way to determine that the `const { getItem }` access was unsafe.
```

Comment:
Well - no robust and generalizable way.... you could still do this syntactically with explicitly listing builtins and members that aren't safe to separate from their object (😬), and I wouldn't be shocked if there are lint rules in the wild that do more or less that